src/misc/programs.cpp

```diff
@@ -376,6 +376,89 @@ class CONFIG final : public Program {
 	}
 };
 
+static char msg[200];
+const char *SetProp(std::vector<std::string> &pvars) {
```

Comment:
Is there a reason for not just returning a `std::string`? It would simplify things a lot, and be safer.

Reply:
Functions like safe_sprintf do not work on std::string. So I returned the char array as is. Let me know if you have solution to this.

Reply:
Yeah, It was one or the other, I wrote the other suggestions before I thought about `std::string`. Still think `std::string` is the better option if you don't mind changing it, otherwise ok.

Reply:
The function ``safe_sprintf`` is used several times, which only work on char and not std::string. If there is an efficient way to implement the function with std::string then great.